multidownloader/evm_multidownloader.go

```diff
@@ -0,0 +1,607 @@
+package multidownloader
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+
+	jRPC "github.com/0xPolygon/cdk-rpc/rpc"
+	aggkitcommon "github.com/agglayer/aggkit/common"
+	"github.com/agglayer/aggkit/db/compatibility"
+	dbtypes "github.com/agglayer/aggkit/db/types"
+	"github.com/agglayer/aggkit/etherman"
+	ethermanblocknotifier "github.com/agglayer/aggkit/etherman/block_notifier"
+	ethermantypes "github.com/agglayer/aggkit/etherman/types"
+	"github.com/agglayer/aggkit/log"
+	"github.com/agglayer/aggkit/multidownloader/storage"
+	mdrtypes "github.com/agglayer/aggkit/multidownloader/types"
+	aggkittypes "github.com/agglayer/aggkit/types"
+	"github.com/ethereum/go-ethereum/core/types"
+	ethrpc "github.com/ethereum/go-ethereum/rpc"
+)
+
+const safeMode = true
+
+// const unsafeMode = false
+
+type EVMMultidownloader struct {
+	log                  aggkitcommon.Logger
+	cfg                  Config
+	ethClient            aggkittypes.BaseEthereumClienter
```

Comment:
Do we need these two separations to `ethClient` and `rpcClient` since `aggkittypes.EthClienter` already implements both interfaces?